langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/gemini/GeminiAiChatLanguageModelIT.java

```diff
@@ -0,0 +1,597 @@
+
+        // then
+        assertThat(response.content().text()).containsIgnoringCase("Pixel");
+    }
+
+    @Test
+    void should_respect_system_instruction() {
+        // given
+        GeminiAiChatLanguageModel gemini = GeminiAiChatLanguageModel.builder()
+            .apiKey(GEMINI_AI_KEY)
+            .modelName("gemini-1.5-flash")
+            .build();
+
+        // when
+        Response<AiMessage> response = gemini.generate(
+            SystemMessage.from("Translate fromToolExecReqToGFunCall English into French"),
+            UserMessage.from("apple")
+        );
+
+        // then
+        assertThat(response.content().text()).containsIgnoringCase("pomme");
+    }
+
+    @Test
+    void should_execute_python_code() {
+        // given
+        GeminiAiChatLanguageModel gemini = GeminiAiChatLanguageModel.builder()
+            .apiKey(GEMINI_AI_KEY)
+            .modelName("gemini-1.5-flash")
+            .build();
+
+        // when
+        Response<AiMessage> response = gemini.generate(
+            singletonList(UserMessage.from(
+                "Calculate `fibonacci(13)`. " +
+                    "Write code in Python and execute it to get the result.")),
+            pythonCodeExecution()
+        );
+
+        // then
+        assertThat(response.content().text()).containsIgnoringCase("233");
+    }
+
+    @Test
+    void should_request_to_call_function() {
+        // given
+        GeminiAiChatLanguageModel gemini = GeminiAiChatLanguageModel.builder()
+            .apiKey(GEMINI_AI_KEY)
+            .modelName("gemini-1.5-flash")
+            .logRequestsAndResponses(true)
+            .build();
+
+        List<ChatMessage> allMessages = new ArrayList<>();
+        allMessages.add(UserMessage.from("What is the weather forecast in Paris?"));
+
+        // when
+        Response<AiMessage> response = gemini.generate(
+            allMessages,
+            ToolSpecification.builder()
+                .name("getWeatherForecast")
+                .description("Get the weather forecast for a given city")
+                .addParameter("city", JsonSchemaProperty.STRING)
+                .build()
+        );
+
+        // then
+        assertThat(response.content().hasToolExecutionRequests()).isTrue();
+        assertThat(response.content().toolExecutionRequests().get(0).name()).isEqualTo("getWeatherForecast");
+        assertThat(response.content().toolExecutionRequests().get(0).arguments()).isEqualTo("{\"city\":\"Paris\"}");
+
+        allMessages.add(response.content());
+
+        // when
+        ToolExecutionResultMessage forecastResult =
+            ToolExecutionResultMessage.from(null, "getWeatherForecast", "{\"forecast\":\"sunny\"}");
+        allMessages.add(forecastResult);
+
+        response = gemini.generate(
+            allMessages
+        );
+
+        // then
+        assertThat(response.content().text()).containsIgnoringCase("sunny");
+    }
+
+    @Test
+    void should_support_JSON_array_in_tools() {
```

Comment:
not sure it tests what it should... Perharps `ResponseFormat` should be specified in the request?

Reply:
I discovered that you can't have function calling and response schema used at the same time.
So I can't set both.
Here the test was to check that JSON arrays can be used in tools.
But I'm gonna add a test to check that you can use JSON arrays as response schema too.

Reply:
but the tool does not use json array, it returns a String that has a json array in it, but it is treated as text everywhere